sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlParser.scala

```diff
@@ -423,19 +396,22 @@ class StaxXmlParser(
                 }
               } else {
                 StaxXmlParserUtils.skipChildren(parser)
+                StaxXmlParserUtils.consumeNextEndElement(parser)
```

Comment:
Why not skip EndElement in skipChildren itself?

Reply:
`skipChildren` is called recursively. We just wanted to skip EndElement after skipping all its children